CC0LRB/Todo/Program.cs

```diff
@@ -0,0 +1,235 @@
+                xs.Serialize(f, list);
+            }
+
+            using (var f = File.OpenRead("test.xml"))
+            {
+                List<Teendo> vissza = xs.Deserialize(f) as List<Teendo>;
+            }
+            #endregion
+            // Szerializacio VEGE
+
+            // Linq KEZDETE
+            #region Linq
+            List<User> Users = new List<User>
+            {
+            new User
+            {
+                UserName = "Alice",
+                TeendoCount = 120,
+            },
+            new User
+            {
+                UserName = "Bob",
+                TeendoCount = 30,
+            },
+            new User
+            {
+                UserName = "Istv�n",
+                TeendoCount = 69,
+            },
+            };
+
+            var bela = new User
+            {
+                UserName = "B�la",
+                TeendoCount = 51,
+            };
+
+            bela.IsNoob();
+
+            //LINQ - Language integrated Query
+            //lambda syntax
+            var max = Users.Max(user => user.TeendoCount);
+
+            var Noobs = from user in Users
+                        where user.IsNoob()
+                        orderby user.TeendoCount descending
+                        select user.UserName;
+
+            foreach (var user in Noobs.Take(2))
+            {
+                Console.WriteLine(user);
+            }
+            #endregion
+            // Linq VEGE
+
+            // Generikus KEZDETE
+            #region generikus
+            var teszt1 = Generikus<int>.Letrehoz(22);
+            var teszt2 = Generikus<double>.Letrehoz(33.2);
+            var teszt3 = Generikus<char>.Letrehoz('A');
+            //az alabbi hibat fog dobni, mert a string osztaly!
+            //Generikus<string> teszt4 = Generikus.Letrehoz("Teszt");
+            #endregion
+            // Generikus VEGE
+
+            // Aszinkron resz KEZDETE
+            #region aszinkron
+            // A task magja �gy egy async Action delegate.
+            Task.Run(async () =>
+             {
+                 Console.WriteLine("Async elott");
+                 await PrintCurrentTimeAsync();
+                 Console.WriteLine("Async utan");
+            });
+            #endregion
+            // Aszinkron resz VEGE
+
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        // Aszinkron Task method
+        private static async Task PrintCurrentTimeAsync()
+        {
+            Console.WriteLine(DateTime.Now);
+            await Task.Delay(2000);
+            Console.WriteLine(DateTime.Now);
+        }
+
+        // record (immutable type) KEZDETE
+        #region record
+        private record NapiTeendo(string micsoda);
+        private static NapiTeendo[] data = new NapiTeendo[]
+        {
+            new NapiTeendo("Ezt meg meg kell csinalni"),
+            new NapiTeendo("Ezt is meg kell csinalni")
+        };
+        #endregion
+        // record (immutable type) VEGE
```

Comment:
Ez így picit Hack the system, mivel ezeket a funkciókat bele is lehetett volna építeni az alkalmazásba némi kreativitással.